packages/slate-react/src/components/editor.js

```diff
@@ -71,100 +72,61 @@ class Editor extends React.Component {
-      }
-    })
+  tmp = {
+    updates: 0,
+    resolves: 0,
+    change: undefined,
   }
 
-  /**
-   * When the `props` are updated, create a new `Stack` if necessary and run
-   * `onChange` to ensure the value is normalized.
-   *
-   * @param {Object} props
-   */
-
-  componentWillReceiveProps = props => {
-    let { schema, stack } = this
-
-    // Increment the updates counter as a baseline.
-    this.tmp.updates++
-
-    // If the plugins or the schema have changed, we need to re-resolve the
-    // plugins, since it will result in a new stack and new validations.
-    if (
-      props.plugins != this.props.plugins ||
-      props.schema != this.props.schema
-    ) {
-      const plugins = this.resolvePlugins(props.plugins, props.schema)
-      stack = Stack.create({ plugins })
-      schema = Schema.create({ plugins })
-      this.setState({ schema, stack })
-
-      // Increment the resolves counter.
-      this.tmp.resolves++
-
-      // If we've resolved a few times already, and it's exactly in line with
-      // the updates, then warn the user that they may be doing something wrong.
-      if (this.tmp.resolves > 5 && this.tmp.resolves == this.tmp.updates) {
-        logger.warn(
-          'A Slate <Editor> is re-resolving `props.plugins` or `props.schema` on each update, which leads to poor performance. This is often due to passing in a new `schema` or `plugins` prop with each render by declaring them inline in your render function. Do not do this!'
-        )
-      }
-    }
-
-    // Run `onChange` on the passed-in value because we need to ensure that it
-    // is normalized, and queue the resulting change.
-    const change = props.value.change()
-    stack.run('onChange', change, this)
-    this.queueChange(change)
-    this.setState({ value: change.value })
-  }
+  state = {}
 
   /**
    * When the component first mounts, flush any temporary changes,
    * and then, focus the editor if `autoFocus` is set.
    */
 
-  componentDidMount = () => {
+  componentDidMount() {
     this.flushChange()
 
     if (this.props.autoFocus) {
       this.focus()
     }
+
+    this.tmp.updates++
   }
 
   /**
    * When the component updates, flush any temporary change.
    */
 
-  componentDidUpdate = () => {
+  componentDidUpdate(prevProps) {
     this.flushChange()
+    if (prevProps === this.props) return
+    // Increment the updates counter as a baseline.
+    this.tmp.updates++
+
+    // If we've resolved a few times already, and it's exactly in line with
+    // the updates, then warn the user that they may be doing something wrong.
+    if (this.tmp.resolves > 5 && this.tmp.resolves == this.tmp.updates) {
+      logger.warn(
+        'A Slate <Editor> is re-resolving `props.plugins` or `props.schema` on each update, which leads to poor performance. This is often due to passing in a new `schema` or `plugins` prop with each render by declaring them inline in your render function. Do not do this!'
```

Comment:
how about making that === 5 so you only warn once?

Reply:
This warning only appears in dev mode. We want the warn appears in each update, so the developer cannot miss it.